ksp_plugin/flight_plan.hpp

```diff
@@ -48,19 +52,34 @@ class FlightPlan {
                  generalized_adaptive_step_parameters);
   virtual ~FlightPlan() = default;
 
+  // Construction parameters.
   virtual Instant initial_time() const;
-  virtual Instant actual_final_time() const;
   virtual Instant desired_final_time() const;
 
+  // End time of the last coast.  If this is less than |desired_final_time()|,
+  // there is at least an anomalous manœuvre.
+  virtual Instant actual_final_time() const;
+
+  // The number of manœuvres in the flight plan.
   virtual int number_of_manœuvres() const;
-  // |index| must be in [0, number_of_manœuvres()[.
+
+  // The number of manœuvres at the end of flight plan that are anomalous, i.e.,
+  // lead to their burn or a subsequent trajectory failing to integrate.  Note
+  // that this returns 1 for a flight plan without manœuvres if the first coast
+  // fails to integrate.  The functions that change manœuvres may change the
+  // number of anomalous manœuvres.
+  virtual int number_of_anomalous_manœuvres() const;
```

Comment:
If`number_of_anomalous_manœuvres` can be less than `number_of_manœuvres`, there is a naming problem.

Let us define an anomalous manœuvre as one whose burn integration cannot be started (because it starts after an anomalous segment). This becomes
`(anomalous_segments_ - 1) / 2`, rather than ~~`(anomalous_segments_ + 1) / 2`~~ (0 for 1 or 2 anomalous segments, 1 for 3 because the segment preceding the last burn is anomalous).

This is only used outside of tests as `number_of_anomalous_manœuvres() <= 1`, which should be changed to `number_of_anomalous_manœuvres() == 1`: the existing semantics are that there are no anomalous manœuvres, that is, the integration of their burn can at least start, and they are editable in the UI.

Reply:
Done.